pkg/controller/instance/delta/delta_test.go

```diff
@@ -458,3 +458,121 @@ func TestCompare_EmptyMaps(t *testing.T) {
 		})
 	}
 }
+
+func TestCompare_EmptyArraysAndNulls(t *testing.T) {
+	tests := []struct {
+		name     string
+		desired  *unstructured.Unstructured
+		observed *unstructured.Unstructured
+		wantDiff bool
+	}{
+		{
+			name: "empty array equals null in nested structure",
+			desired: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"template": map[string]interface{}{
+							"containers": []interface{}{
+								map[string]interface{}{
+									"env": []interface{}{},
+								},
+							},
+						},
+					},
+				},
+			},
+			observed: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"template": map[string]interface{}{
+							"containers": []interface{}{
+								map[string]interface{}{
+									"env": nil,
```

Comment:
minor: a test that asserts that `map[string]interface{}` with no key at all (no `env`) _does_ differ from one that has an `env` key whose value is nil?

I think that's the expected behavior, right?

Reply:
yes. added.